### DISCUSSION TITLE
Extending existing component

### DISCUSSION THREAD

**Discussion by amoscatelli** [keycloakify/keycloakify#81] [Q&A]
Hi everybody.

First of all :

Thank you for this project.
I do love Keycloak but coding FTL is really a pain and I think what are you doing here is great.

So let's talk about business :
I am no react programmer, I am more an Angular guy (but I like trying new things so no problem about this).
What I have managed to do so far on my own ?
I created a react project, builded a keycloak theme with keycloakify and testing both live (yarn test) and inside Docker image via your .sh and customized the style of a specific component.
I kind think of understand how to create a completely new component/page/ftl also.

What I am missing is the more straightful way to extend modify an existing component.

For example Login.tsx : 
https://github.com/InseeFrLab/keycloakify/blob/main/src/lib/components/Login.tsx

What if I want to add a button to the formNode or the infoNode ?
Is there some way to reuse the rest of the Login const and only overriding the formNode part ?

Thank you in advance
I am sorry to bother you with this but it seems I was unable to find something similar in your examples.

**Comment by garronej** [comment]
Hi @amoscatelli,  

Thank you very much for your display of appreciation!  
Keycloakify enables you to customise things as much as you want.  
Did you had a look at [this branch](https://github.com/garronej/keycloakify-demo-app/tree/look_and_feel) of the demo setup?  
You can also checkout [this production example](https://github.com/InseeFrLab/onyxia-web/tree/main/src/ui/components/KcApp).  

Let me know if it help,  

Best

**Reply by amoscatelli** [reply]
hi @garronej !

Thank you.
So it seems to me there is no way of 'extending'.

All I see in those examples is 'redefining' :
https://github.com/garronej/keycloakify-demo-app/blob/look_and_feel/src/KcApp/Register.tsx

I read in the comment 'This is a copy paste' ... I was trying to avoid a copy paste, that was the point :) 

Well, it's fine. I just wanted to be sure I understood correctly.

**Reply by garronej** [reply]
Hi!

If tou re use the Template.tsx from component from Keycloakify you do not need to mess with anything besides the form.  

If you just want to add some fields it can be done from the admin console using User profile and "register-user-profile.ftl" (instead of "register.ftl"). 

Hope it helps,

Best
